packages/charts/src/chart_types/xy_chart/state/chart_state.interactions.test.tsx

```diff
@@ -1232,4 +1225,112 @@ describe('Clickable annotations', () => {
     const callArgs = onAnnotationClick.mock.calls[0][0];
     expect(callArgs.rects[0].id).toEqual('rect1');
   });
+  test('click with two overlapping rect annotations', () => {
+    const store = MockStore.default({ width: 500, height: 500, top: 0, left: 0 }, 'chartId');
+    const onAnnotationClick = jest.fn<void, any[]>((data: any): void => data);
+    const onAnnotationClickCaller = createOnClickCaller();
+    store.subscribe(() => {
+      onAnnotationClickCaller(store.getState());
+    });
+    MockStore.addSpecs(
+      [
+        MockGlobalSpec.settingsNoMargins({
+          onAnnotationClick,
+        }),
+        MockSeriesSpec.bar({
+          xScaleType: ScaleType.Linear,
+          yScaleType: ScaleType.Linear,
+          xAccessor: 'x',
+          yAccessors: ['y'],
+          data: [
+            { x: 0, y: 2 },
+            { x: 1, y: 3 },
+            { x: 3, y: 6 },
+          ],
+        }),
+        MockAnnotationSpec.rect({
+          id: 'rect1',
+          dataValues: [
+            {
+              coordinates: {
+                x0: 0,
+                x1: 1,
+                y0: 0,
+                y1: 4,
+              },
+              details: 'details about this annotation',
+            },
+          ],
+          style: { fill: 'red' },
+        }),
+        MockAnnotationSpec.rect({
+          id: 'rect2',
+          dataValues: [
+            {
+              coordinates: {
+                x0: 1,
+                x1: 2,
+                y0: 1,
+                y1: 5,
+              },
+              details: 'details about this other annotation',
+            },
+          ],
+          style: { fill: 'blue' },
+        }),
+      ],
+      store,
+    );
+    // the overlap of the blue and red rect
+    store.dispatch(onPointerMove({ x: 200, y: 195 }, 0));
+    store.dispatch(onMouseDown({ x: 200, y: 195 }, 100));
+    store.dispatch(onMouseUp({ x: 200, y: 195 }, 200));
+
+    expect(onAnnotationClick).toBeCalled();
+    const callArgs = onAnnotationClick.mock.calls[0][0];
+    expect(callArgs.rects[0].id).toEqual('rect2');
```

Comment:
wouldn't you expect both to be in `rects`?

```suggestion
    expect(callArgs.rects[0].id).toEqual('rect1');
    expect(callArgs.rects[1].id).toEqual('rect2');
```

Reply:
I attempted a large scale refactoring but then backtracked in 495fc1b. I think it might introduce a breaking change in what is returned for tooltips so it might be better for a new PR.